hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/response/TestCleanupTableInfoResponses.java

```diff
@@ -0,0 +1,331 @@
+      Mockito.lenient().doAnswer(answer).when(mockedTable).deleteWithBatch(any(), any());
+      mockedTableMap.put(tableMap.get(table), mockedTable);
+      tableMap.put(table, mockedTable);
+    }
+    for (Field f:omMetadataManager.getClass().getDeclaredFields()) {
+      boolean isAccessible = f.isAccessible();
+      f.setAccessible(true);
+      try {
+        if (Table.class.isAssignableFrom(f.getType())) {
+          f.set(omMetadataManager,
+              mockedTableMap.getOrDefault(f.get(omMetadataManager),
+                  (Table)f.get(omMetadataManager)));
+        }
+      } finally {
+        f.setAccessible(isAccessible);
+      }
+    }
+  }
+  private <T> T getMockObject(Class<T> type,
+                              Map<Class, Object> instanceMap,
+                              OzoneManagerProtocolProtos.Status status,
+                              boolean isFSO) {
+    T instance =  Mockito.mock(type, invocationOnMock -> {
+      try {
+        return createInstance(invocationOnMock.getMethod().getReturnType(),
+            instanceMap, status, isFSO);
+      } catch (Throwable e) {
+        throw new RuntimeException(e);
+      }
+    });
+    instanceMap.put(type, instance);
+    return instance;
+  }
+  private List<Field> getFields(Class<?> type) {
+    List<Field> fields = new ArrayList<>();
+    while (type != Object.class) {
+      fields.addAll(Arrays.asList(type.getDeclaredFields()));
+      type = type.getSuperclass();
+    }
+    return fields;
+  }
+  private Object createInstance(Class<?> type,
+                                Map<Class, Object> instanceMap,
+                                OzoneManagerProtocolProtos.Status status, boolean isFSO) {
+    if (instanceMap.containsKey(type)) {
+      return instanceMap.get(type);
+    }
+    Object instance = null;
+    if (type.isArray()) {
+      return Array.newInstance(type.getComponentType(), 0);
+    } else if (type.equals(Void.TYPE)) {
+      return null;
+    } else if (type.isPrimitive()) {
+      if (Boolean.TYPE.equals(type)) {
+        return true;
+      } else if (Character.TYPE.equals(type)) {
+        return 'a';
+      } else if (Byte.TYPE.equals(type)) {
+        return 0xFF;
+      } else if (Short.TYPE.equals(type)) {
+        return (short) 0;
+      } else if (Integer.TYPE.equals(type)) {
+        return 0;
+      } else if (Long.TYPE.equals(type)) {
+        return (long) 0;
+      } else if (Float.TYPE.equals(type)) {
+        return (float) 0.0;
+      } else if (Double.TYPE.equals(type)) {
+        return 0.0;
+      } else if (Void.TYPE.equals(type)) {
+        return null;
+      }
+    } else if (type.equals(OzoneManagerProtocolProtos.OMResponse.class)) {
+      return OzoneManagerProtocolProtos.OMResponse.newBuilder()
+          .setStatus(status).buildPartial();
+    } else if (type.equals(BucketLayout.class)) {
+      return isFSO ? BucketLayout.FILE_SYSTEM_OPTIMIZED
+          : BucketLayout.DEFAULT;
```

Comment:
It would be better to explicitly define the bucket layout here. For instance `LEGACY` or `OBJECT_STORE` instead of `DEFAULT`.